src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,187 @@
+        //creates a child for the encoders
+        addChild(liftEncoder);
+        addChild(topLimitSwitch);
+        addChild(bottomLimitSwitch);
+
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_P, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_I, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_D, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_F, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.LIFT_UP_DOWN_INCRIMENT, 1);
+    }
+
+    protected double returnPIDInput() {
+        return liftEncoder.getDistance();
+    }
+
+    protected void usePIDOutput(double output) {
+        if (topLimitSwitch.get() == true && output > 0) {
+            setSetpoint(LiftHeight.kMaxHeight.get());
+            return;
+        }
+        if (bottomLimitSwitch.get() == true && output < 0) {
+            setSetpoint(LiftHeight.kFloor.get());
+            return;
+        }
+        liftDrive.set(output);
+    }
+
+    //gear changes
+    public void shiftToHighGear() {
+        liftTransmission.set(Value.kForward);
+    }
+
+    public void shiftToLowGear() {
+        liftTransmission.set(Value.kReverse);
+    }
+
+    //does not do anything
+    public void initDefaultCommand() {
+    }
+
+    public Command GoToHeight(LiftHeight height, boolean isHighGear) {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                if (isHighGear == true) {
+                    shiftToHighGear();
+                } else
+                    shiftToLowGear();
+
+                setSetpoint(height.get());
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        };
+    }
+
+    public Command GoUp() {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                setSetpoint(getSetpoint()
+                        + Preferences.getInstance().getDouble(RobotMap.Preferences.LIFT_UP_DOWN_INCRIMENT, 10));
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        };
+    }
+
+    public Command GoDown() {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                setSetpoint(getSetpoint()
+                        - Preferences.getInstance().getDouble(RobotMap.Preferences.LIFT_UP_DOWN_INCRIMENT, 10));
+            }
+
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        };
+    }
+
+    public Command ClimbUp() {
+        return new CommandChain("Climb Up").then(GoToHeight(LiftHeight.kClimb, true)).then(GoToHeight(LiftHeight.kScaleLow, false));              GoToHeight(LiftHeight.kScaleLow, false));
```

Comment:
There's some weird  GoToHeight(LiftHeight.kScaleLow, false)); thing at the end of this line. Guessing you just missed it.